### DISCUSSION TITLE
[RFC] Minimum required PHP version bump to 8.1

### DISCUSSION THREAD

**Discussion by mychidarko** [leafsphp/leaf#245] [Polls]
Leaf 3 started out with support for PHP 7.1 and above because most frameworks had dropped support for PHP 7.4 and lower. We wanted to give developers a framework that they could use with their existing projects without having to upgrade their PHP version. However, PHP 8 has been out for a while now, and it comes with a lot of new features and improvements that we would like to take advantage of. Our most recent metrics show just about 6% of our users are on PHP 7.4, 3% use PHP 8.0, with no devs on PHP 7.3 and below. We believe that it's time to make PHP 8.1 the minimum requirement for Leaf and take advantage of the improvements it offers. We would like to hear your thoughts on this change and how it might affect your projects.

**Comment by wilsenhc** [comment]
This would greatly benefit the framework as it would provide a lot of new features and improvements from PHP 8.1 as a base/minimum version. Yes, go ahead with this change!

**Comment by mychidarko** [comment]
Thanks for all the input, we will go ahead with this change
